core/src/main/java/brooklyn/catalog/internal/CatalogDo.java

```diff
@@ -254,8 +270,10 @@ CatalogDo addCatalog(CatalogDto child) {
     public synchronized void addToClasspath(String ...urls) {
         if (dto.classpath == null)
             dto.classpath = new CatalogClasspathDto();
-        for (String url: urls)
-            dto.classpath.addEntry(url);
+        for (String url: urls) {
+            if (url!=null)
```

Comment:
Let's hope we never hit this with `url == null`. I'd expect (at least some times) that will be caused by a programming error or missing resource, which we should tell the caller. This code will now ignore (without any indication to the user).

This relates to a more general discussion of how we should handle nulls. I think we should avoid nulls where feasible and accept them where required (e.g. getAttribute() and getConfig() can always return null). But I don't feel strongly enough to say more at this time!